src/Mollie/WC/Helper/PaymentMethodsIconUrl.php

```diff
@@ -39,7 +62,130 @@ protected function fallToAssets($paymentMethodName)
+
+        return  $this->writeToSvgFile($actual);
+    }
+
+    /**
+     * Is the customization of credit card icons enabled?
+     *
+     * @return bool
+     */
+    protected function enabledCreditcardOptions()
+    {
+        $creditcardSettings = get_option(
+            'mollie_wc_gateway_creditcard_settings'
+        );
+        if (!isset($creditcardSettings['mollie_creditcard_icons_enabler'])) {
+            return false;
+        }
+        return wc_string_to_bool(
+            $creditcardSettings['mollie_creditcard_icons_enabler']
+        );
+    }
+
+    /**
+     * @return array Array containing the credit cards names enabled in settings
+     *               to make customization of checkout icons
+     */
+    protected function enabledCreditcards()
+    {
+        $optionLexem = self::MOLLIE_CREDITCARD_ICONS;
+        $creditcardsAvailable = self::AVAILABLE_CREDITCARD_ICONS;
+        $svgFileName = self::SVG_FILE_EXTENSION;
+        $enabledCreditcards = [];
+
+        $creditcardSettings = get_option('mollie_wc_gateway_creditcard_settings');
+        foreach ($creditcardsAvailable as $card) {
+            if ($creditcardSettings[$optionLexem . $card] === 'yes') {
+                $enabledCreditcards[] = $card . $svgFileName;
+            }
+        }
+
+        return $enabledCreditcards;
+    }
+
+    /**
+     * Method to write the svg file with the composed svg
+     * Can be used to reset the file to empty
+     *
+     * @param string $content svg content to compose the image if empty will
+     *                        reset the file
+     *
+     * @return string the url to the written file
+     */
+    protected function writeToSvgFile($content = '')
+    {
+        $uploads = wp_upload_dir();
+        file_put_contents(
+            $uploads['basedir'] . '/compositeCards.svg',
+            $content
+        );
+        return $uploads['baseurl'] . '/compositeCards.svg';
+    }
+
+    /**
+     * @param array $enabledCreditCards Array of the selected cards to compose
+     *                                  the image with
+     *
+     * @return string Newly composed svg string
+     */
+    protected function buildSvgComposed($enabledCreditCards)
+    {
+        $assetsImagesPath
+            = '/app/public/wp-content/plugins/WooCommerce-Mollie/public/images/Creditcard_issuers/';
+        $cardWidth = self::CREDIT_CARD_ICON_WIDTH;
+        $cardsNumber = count($enabledCreditCards);
+        $cardsWidth = $cardWidth * $cardsNumber;
+        $cardPositionX = 0;
+        $actual
+            = "<svg width=\"{$cardsWidth}\" height=\"24\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">";
+        foreach ($enabledCreditCards as $creditCard) {
+            $actual .= $this->positionSvgOnX(
+                $cardPositionX,
+                file_get_contents(
+                    $assetsImagesPath . $creditCard
+                )
+            );
```

Comment:
What if it's not possible to read the content of the file for some reason?

We want to probably skip the entire build process for the `$actual` or just skip one?